ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1240 @@
+      return new WithQuery(node.getLocation().get(),
+          node.getName(),
+          (Query) process(node.getQuery(), context),
+          node.getColumnNames());
+    } else {
+      return new WithQuery(node.getName(),
+          (Query) process(node.getQuery(), context),
+          node.getColumnNames());
+    }
+
+  }
+
+  protected Node visitSelect(Select node, Object context) {
+    if (node.getLocation().isPresent()) {
+      return new Select(node.getLocation().get(),
+          node.isDistinct(),
+          node.getSelectItems()
+              .stream()
+              .map(selectItem -> (SelectItem) process(selectItem, context))
+              .collect(Collectors.toList())
+      );
+    } else {
+      return new Select(node.isDistinct(),
+          node.getSelectItems()
+              .stream()
+              .map(selectItem -> (SelectItem) process(selectItem, context))
+              .collect(Collectors.toList())
+      );
+    }
+
+  }
+
+  protected Node visitRelation(Relation node, Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitQueryBody(QueryBody node, Object context) {
+    return visitRelation(node, context);
+  }
+
+  protected Node visitQuerySpecification(QuerySpecification node, Object context) {
+    Optional<WindowExpression> windowExpression = node.getWindowExpression().isPresent()
+        ? Optional.ofNullable((WindowExpression) process(node.getWindowExpression().get(), context))
+        : Optional.empty();
+    Optional<Expression> where = node.getWhere().isPresent()
+        ? Optional.ofNullable((Expression) process(node.getWhere().get(),
+        context))
+        : Optional.empty();
+
+    Optional<GroupBy> groupBy = node.getGroupBy().isPresent()
+        ? Optional.ofNullable((GroupBy) process(node.getGroupBy().get(), context))
+        : Optional.empty();
+
+    Optional<Expression> having = node.getHaving().isPresent()
+        ? Optional.ofNullable((Expression) process(node.getHaving().get(), context))
+        : Optional.empty();
+
+    if (node.getLocation().isPresent()) {
+      return new QuerySpecification(
+          node.getLocation().get(),
+          (Select) process(node.getSelect(), context),
+          (Relation) process(node.getInto(), context),
+          node.isShouldCreateInto(),
+          (Relation) process(node.getFrom(), context),
+          windowExpression,
+          where,
+          groupBy,
+          having,
+          node.getLimit()
+      );
+    } else {
+      return new QuerySpecification(
+          (Select) process(node.getSelect(), context),
+          (Relation) process(node.getInto(), context),
+          node.isShouldCreateInto(),
+          (Relation) process(node.getFrom(), context),
+          windowExpression,
+          where,
+          groupBy,
+          having,
+          node.getLimit()
+      );
+    }
+
+  }
+
+  protected Node visitSetOperation(SetOperation node, Object context) {
```

Comment:
Can you rely on the default behavior here? Similar comment for other no-op visit functions except for visitNode

Reply:
Yes, this is more for Queries than other commands such as `SET` or `PRINT`.